src/Infrastructure/LeanCode.Serialization/JsonConverters/JsonLaxDateOnlyConverter.cs

```diff
@@ -1,16 +1,26 @@
+using System.Buffers;
 using System.Globalization;
+using System.Text;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 
 namespace LeanCode.Serialization;
 
 public class JsonLaxDateOnlyConverter : JsonConverter<DateOnly>
 {
+    private static readonly UTF8Encoding encoder = new(false);
+
     public override DateOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        return reader.GetString() is string s
-            ? DateOnly.FromDateTime(DateTime.Parse(s, CultureInfo.InvariantCulture))
-            : default;
+        var value = reader.ValueSpan;
+        Span<char> chars = stackalloc char[value.Length];
+        encoder.GetChars(value, chars);
+        if (DateTime.TryParse(chars ,out DateTime date))
```

Comment:
Won't this attempt to parse a potentially escaped raw JSON value? Please make sure that the following succeeds:

```csharp
JsonSerializer.Deserialize<DateOnly>(@"""\u0032\u0030\u0032\u0032-\u0030\u0032-\u0030\u0032""")
```

.NET's builtin converter properly recognizes this as `2022-02-02`. please make sure ours does as well.